x-pack/plugin/ilm/qa/multi-node/src/test/java/org/elasticsearch/xpack/ilm/TimeSeriesLifecycleActionsIT.java

```diff
@@ -1120,26 +1120,26 @@ public void testRolloverStepRetriesUntilRolledOverIndexIsDeleted() throws Except
         // {@link org.elasticsearch.xpack.core.ilm.ErrorStep} in order to retry the failing step. As {@link #assertBusy}
         // increases the wait time between calls exponentially, we might miss the window where the policy is on
         // {@link WaitForRolloverReadyStep} and the move to `attempt-rollover` request will not be successful.
-        waitUntil(() -> {
+        assertThat(waitUntil(() -> {
```

Comment:
I'd consider using `assertTrue`, we have long body here, it's hard to tell what we asserting here. The same applies to all instances below

Reply:
It could be changed to just `assertBusy(() -> {...}, 30, TimeUnit.SECONDS)` also and use assertions to signal success/failure

Reply:
I've changed the `assertThat(waitUntil(), is(true))` to `assertTrue(waitUntil())` but the reason to use `waitUntil` as opposed to `assertBusy` is to avoid having this test flake. We are asserting that the failed step and the retry count are bound to particular values. The fact that we want to check the failed step is what we expect it to be means we have to *catch* ILM in the ERROR step, but with retryable steps ILM keeps moving back and forth between the failing step (on retry) and the ERROR step (when the step execution fails).

`assertBusy` increases the wait time exponentially once it gets to 1 second, while `waitUntil` keeps it at 1 second. This means we're bound to catch ILM in the ERROR step using `waitUntil` (but with assertBusy we probe the ILM state only at seconds 1, 2, 4, 8 and 16)

Reply:
Ahh okay, I see the difference between them now, in that case I agree that `assertTrue` would be better than `assertThat` in this case